lighthouse-core/report/html/renderer/details-renderer.js

```diff
@@ -536,24 +536,50 @@ class DetailsRenderer {
       return null;
     }
 
+    /** @param {HTMLElement} el */
+    function addDevToolsData(el) {
+      el.classList.add('lh-source-location');
+      el.setAttribute('data-source-url', item.url);
+      // DevTools expects zero-indexed lines.
+      el.setAttribute('data-source-line', String(item.line));
+      el.setAttribute('data-source-column', String(item.column));
+      return el;
+    }
+
     // Lines are shown as one-indexed.
-    const line = item.line + 1;
-    const column = item.column;
+    const line = (item.original ? item.original.line : item.line) + 1;
+    const column = item.original ? item.original.column : item.column;
 
-    let element;
+    // First two cases: url is from network.
```

Comment:
it seems like there's gotta be a better way to structure these conditionals.

Reply:
this is much better thx